mender-update/context/context.cpp

```diff
@@ -371,6 +376,59 @@ error::Error MenderContext::CommitArtifactData(
 	});
 }
 
+expected::ExpectedBool MenderContext::MatchesArtifactDepends(const artifact::HeaderInfo &hdr_info) {
+	auto ex_dev_type = GetDeviceType();
+	if (!ex_dev_type) {
+		return expected::unexpected(ex_dev_type.error());
+	}
+	auto ex_provides = LoadProvides();
+	if (!ex_provides) {
+		return expected::unexpected(ex_provides.error());
+	}
+	auto &provides = ex_provides.value();
+	if (provides.find("artifact_name") == provides.end()) {
+		return expected::unexpected(MakeError(ValueError, "Missing artifact_name value"));
+	}
+	if (provides.find("artifact_group") == provides.end()) {
```

Comment:
`artifact_group` can be (and usually is) missing. In fact, it has rarely been used, because we introduced it early, but never used it for its intended purpose. But since it's in the database we have to keep supporting it.

`artifact_name` is guaranteed to be present though.